### PR TITLE
Turtlebot Tutorial update

### PR DESCRIPTION
**Pull request by gtep96**
Requesting review

### CODE REVIEW

File: docs_tutorials/turtlebot/turtlebot.md [hunk 1]
```diff
@@ -5,231 +5,149 @@ sidebar_label: 'TurtleBot'
...
 
-```python
-citros.xy_plot(ax1, 
-               topic_name = '/odom', 
-               var_x_name = 'data.pose.pose.position.x',
-               var_y_name = 'data.pose.pose.position.y',
-               sids = [0,1,2], 
-               x_label = 'x, m', y_label = 'y, m', title_text = 'XY path plot for sids ##0-2')
-```
 As you can see, the travelled trajectory varies for different sids:
```
Comment (mariaBer) [inline docs_tutorials/turtlebot/turtlebot.md:148]: what is sids?

**Review by mariaBer — CHANGES_REQUESTED** [review]

File: docs_tutorials/turtlebot/turtlebot.md [hunk 2]
```diff
@@ -5,231 +5,149 @@ sidebar_label: 'TurtleBot'
...
-
-* **Orientation and Navigation**: The TurtleBot3 robot is capable of orienting itself effectively within a prepared TurtleBot3 world. It utilizes the simulated lidar module to gather environmental data, allowing it to make informed decisions about its navigation path.
+## Overview
+This project is designed to wrap the official TurtleBot3 simulation [example](https://emanual.robotis.com/docs/en/platform/turtlebot3/simulation/#gazebo-simulation) and integrate it with a simple Autonomous Collision Avoidance node. The node provides ROS 2 parameters that allow users to adjust various TurtleBot3 specifications, enabling the analysis of how these changes affect collision avoidance behavior. 
 
 All project installation, code overview and usage details are also available on the project's [GitHub page](https://github.com/citros-garden/turtlebot3).
 
 ![png](img/turtlebot3_0.png "TurtleBot3")
```
[the remark this answers is not in the document]
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md:13]: Done

File: docs_tutorials/turtlebot/turtlebot.md [hunk 3]
```diff
@@ -5,231 +5,149 @@ sidebar_label: 'TurtleBot'
 
 # TurtleBot3 Example Using CITROS
```
[the remark this answers is not in the document]
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md:6]: Done

File: docs_tutorials/turtlebot/turtlebot.md [hunk 4]
```diff
@@ -5,231 +5,149 @@ sidebar_label: 'TurtleBot'
...
 All project installation, code overview and usage details are also available on the project's [GitHub page](https://github.com/citros-garden/turtlebot3).
 
 ![png](img/turtlebot3_0.png "TurtleBot3")
 
+## Prerequisites
+
+1. Please make sure you have all the [necessary softwares](https://citros.io/doc/docs_tutorials/getting_started/#softwares-to-work-with-citros) to work with CITROS installed on your computer.
+2. Install [Visual Studio code](https://code.visualstudio.com/download).
+3. We strongly recommend that you work with [dockers](https://citros.io/doc/docs_tutorials/dockerfile_overview/). However, if you wish to work without dockers, please refer to the .devcontainer [directory](https://github.com/citros-garden/turtlebot3/tree/main/.devcontainer) in project's repo, the dependencies you need are in the ```Dockerfile``` file.
+4. (Optional) Install [FoxGlove](https://docs.foxglove.dev/docs/introduction).
```
[the remark this answers is not in the document]
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md:20]: Done

File: docs_tutorials/turtlebot/turtlebot.md [hunk 5]
```diff
@@ -5,231 +5,149 @@ sidebar_label: 'TurtleBot'
...
+|--|--|--
+separation		|wheel separation	|turtlebot3_gazebo
+radius		|wheel radius	|turtlebot3_gazebo
+check_forward_dist_param		|forward checking distance for Autonomous Collision Avoidance	|turtlebot3_gazebo
+check_side_dist_param		|side checking distance for Autonomous Collision Avoidance	|turtlebot3_gazebo
 
-### CITROS Installation
 
-First of all, to use all the powerful CITROS features the CLI installation is required: follow the instructions on the CITROS CLI [documentation page](https://citros.io/doc/docs_cli).
+This project contains two launch files, but we will use only ```turtlebot3_sim_cont.launch.py```, and the second one launches automatically (it's necessary for publishing states).
```
[the remark this answers is not in the document]
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md:49]: Yes

File: docs_tutorials/turtlebot/turtlebot.md [hunk 6]
```diff
@@ -5,231 +5,149 @@ sidebar_label: 'TurtleBot'
...
-    check_side_dist_param	|turtlebot3_gazebo	|side checking distance for Autonomous Collision Avoidance	
+Now you can see .citros directory in the explorer.
 
-4. Launch files. This project contains two launch files, but we will use only ```turtlebot3_sim_cont.launch.py```, and the second one launches automatically (it's necessary for publishing states).
+## Scenario
+For this example, let's check how the powerfull CITROS Error Analysis (a part of Data Analysis) package works. To do it out, we need to set up parameters and launch CITROS simulation.<br/>
+The ROS 2 node interfaces with the official TurtleBot3 simulation example, leveraging the TurtleBot3 robot model and simulated environment. This integration allows users to observe how the robot behaves in a controlled environment. The node includes a simple Autonomous Collision Avoidance module. This module is responsible for ensuring that the TurtleBot3 avoids collisions with obstacles in its path. It utilizes sensor data, such as simulated lidar readings, to detect obstacles and adjust the robot's trajectory accordingly.<br/>
```
[the remark this answers is not in the document]
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md:66]: ROS 2 Node uses Turtlebot3 simulation with Autonomous Collision Avoidance module connected. Robot drives in the `room` (named turtlebot_world) avoiding obstacles using sensors (e.g. lidars).

File: docs_tutorials/turtlebot/turtlebot.md [hunk 7]
```diff
@@ -5,231 +5,149 @@ sidebar_label: 'TurtleBot'
...
-    |--|--|--
-    separation	|turtlebot3_gazebo	|wheel separation	
-    radius	|turtlebot3_gazebo	|wheel radius	
-    check_forward_dist_param	|turtlebot3_gazebo	|forward checking distance for Autonomous Collision Avoidance	
-    check_side_dist_param	|turtlebot3_gazebo	|side checking distance for Autonomous Collision Avoidance	
+Now you can see .citros directory in the explorer.
 
-4. Launch files. This project contains two launch files, but we will use only ```turtlebot3_sim_cont.launch.py```, and the second one launches automatically (it's necessary for publishing states).
+## Scenario
+For this example, let's check how the powerfull CITROS Error Analysis (a part of Data Analysis) package works. To do it out, we need to set up parameters and launch CITROS simulation.<br/>
```
[the remark this answers is not in the document]
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md:65]: Deleted the first one

File: docs_tutorials/turtlebot/turtlebot.md [hunk 8]
```diff
@@ -5,231 +5,149 @@ sidebar_label: 'TurtleBot'
...
+The ROS 2 parameters provided by this node give users the flexibility to modify various TurtleBot3 specifications. These parameters may include attributes like the robot's size, speed, sensor range, or collision avoidance algorithms. Users can experiment with different parameter values to observe their impact on collision avoidance behavior. The TurtleBot3 robot is capable of orienting itself effectively within a prepared TurtleBot3 world. It utilizes the simulated lidar module to gather environmental data, allowing it to make informed decisions about its navigation path.<br/>
+For this example, let's check how the powerfull CITROS Error Analysis (a part of Data Analysis) package works. To do it out, we need to set up parameters and launch CITROS simulation.<br/>
```
[the remark this answers is not in the document]
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md:68]: As I said, this project doesn't do any research, it was created to show the ability of implementing and using very popular robotic lib Turtlebot to CITROS. In addition, i've added some exapmles of Data Analysis usage to the Notebook. So, the main purpuse of this project is to show the ability to implement the Turtlebot lib to CITROS and to work and to analyse its data. And the secondary is to show some advantages of Data Analysis package. Basicly, this was the main task for all the tutorials - show to the user the ability to work with popular and complex robotic/scientific libs with CITROS.

File: docs_tutorials/turtlebot/turtlebot.md [hunk 9]
```diff
@@ -5,231 +5,149 @@ sidebar_label: 'TurtleBot'
...
->>> citros push
-```
-### Uploading Docker Image to CITROS Database
-We need to build and push a Docker container image to the CITROS server:
-```bash 
->>> citros docker-build-push
-Logging in to docker...
-...
-```
+All the results will be saved under .citros/runs/[simulation_name] [folder].
```
[the remark this answers is not in the document]
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md:120]: Done

File: docs_tutorials/turtlebot/turtlebot.md [hunk 10]
```diff
@@ -5,231 +5,149 @@ sidebar_label: 'TurtleBot'
...
+First ensure that the project has been [built and sourced](https://citros.io/doc/docs_tutorials/getting_started/#build-the-project).
+Now we can launch it locally:
 ```bash 
 >>> citros run -n 'turtlebot3' -m 'local test run'
-? Please choose the simulation you wish to run: 
+? Please choose the simulation you wish to run:
   simulation_robot_state_publisher
 ❯ simulation_turtlebot3_sim_cont
 ```
 Select the launch file by pressing ```Enter``` button and wait for the output in the terminal. To plot the local run results you can use FoxGlove.
```
[the remark this answers is not in the document]
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md:111]: Done

File: docs_tutorials/turtlebot/turtlebot.md [hunk 11]
```diff
@@ -5,231 +5,149 @@ sidebar_label: 'TurtleBot'
...
->>> citros run -n 'turtlebot3' -m 'cloud test run' -r
-? Please choose the simulation you wish to run: 
-  simulation_robot_state_publisher
-❯ simulation_turtlebot3_sim_cont
-```
-Select the launch file (should be the only one here) by pressing ```Enter``` button. Now the simulation is running in the CITROS server, and the results will be automatically uploaded to the CITROS database.
-```bash
-created new batch_id: <your-batch-id-here>. Running on Citros cluster. See https://citros.io/batch/<your-batch-id-here>.
-```
+To plot the local run results you can use [FoxGlove](https://citros.io/doc/docs_tutorials/#visualization-with-foxglove).
```
[the remark this answers is not in the document]
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md:122]: added

File: docs_tutorials/turtlebot/turtlebot.md (hunk 1 above)
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md:148]: done, used simulations insted of sids. Should be more clear now

**Comment by gtep96** [comment]
@mariaBer changes pushed

**Review by mariaBer — CHANGES_REQUESTED** [review]

File: docs_tutorials/turtlebot/turtlebot.md [hunk 12]
```diff
@@ -3,233 +3,150 @@ sidebar_position: 50
...
+|--|--|--
+separation		|wheel separation	|turtlebot3_gazebo
+radius		|wheel radius	|turtlebot3_gazebo
+check_forward_dist_param		|forward checking distance for Autonomous Collision Avoidance	|turtlebot3_gazebo
+check_side_dist_param		|side checking distance for Autonomous Collision Avoidance	|turtlebot3_gazebo
 
-### CITROS Installation
 
-First of all, to use all the powerful CITROS features the CLI installation is required: follow the instructions on the CITROS CLI [documentation page](https://citros.io/doc/docs_cli).
+This project contains two launch files, but we will use only ```turtlebot3_sim_cont.launch.py```, and the second one launches automatically (it's necessary for publishing states).
```
[the remark this answers is not in the document]
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md:48]: Done

File: docs_tutorials/turtlebot/turtlebot.md [hunk 13]
```diff
@@ -3,233 +3,150 @@ sidebar_position: 50
...
-Citros successfully synched with local project.
-You may review your changes via `citros status` and commit them via `citros commit`.
-Intialized Citros repository.
-```
-Now you can see ```.citros``` folder in the explorer.
+|Launch File	|Description	|Package
+|--|--|--
+turtlebot3_sim_cont.launch.py		|Gazebo headless TurtleBot world launch file 	|turtlebot3_gazebo
+robot_state_publisher.launch.py		|Utility launch file for state publishing
+ |turtlebot3_gazebo
```
[the remark this answers is not in the document]
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md:54]: fixed

File: docs_tutorials/turtlebot/turtlebot.md [hunk 14]
```diff
@@ -3,233 +3,150 @@ sidebar_position: 50
...
-    |--|--|--
-    separation	|turtlebot3_gazebo	|wheel separation	
-    radius	|turtlebot3_gazebo	|wheel radius	
-    check_forward_dist_param	|turtlebot3_gazebo	|forward checking distance for Autonomous Collision Avoidance	
-    check_side_dist_param	|turtlebot3_gazebo	|side checking distance for Autonomous Collision Avoidance	
+Now you can see .citros directory in the explorer.
 
-4. Launch files. This project contains two launch files, but we will use only ```turtlebot3_sim_cont.launch.py```, and the second one launches automatically (it's necessary for publishing states).
+## Scenario
+The ROS 2 node interfaces with the official TurtleBot3 simulation example, leveraging the TurtleBot3 robot model and simulated environment. This integration allows users to observe how the robot behaves in a controlled environment. The node includes a simple Autonomous Collision Avoidance module. This module is responsible for ensuring that the TurtleBot3 avoids collisions with obstacles in its path. It utilizes sensor data, such as simulated lidar readings, to detect obstacles and adjust the robot's trajectory accordingly.<br/>
```
[the remark this answers is not in the document]
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md:64]: Done

File: docs_tutorials/turtlebot/turtlebot.md [hunk 15]
```diff
@@ -3,233 +3,150 @@ sidebar_position: 50
...
 
-    |Launch File	|Package	|Description
-    |--|--|--
-    turtlebot3_sim_cont.launch.py	|turtlebot3_gazebo	|Gazebo headless TurtleBot world launch file 	
-    robot_state_publisher.launch.py	|turtlebot3_gazebo	|Utility launch file for state publishing
+Learn more about parameter setup and defining custom functions in [Directory parameter_setups](https://citros.io/doc/docs_cli/structure/citros_structure/#directory-parameter_setups) and [Adding Functions to Parameter Setup](https://citros.io/doc/docs_cli/configuration/config_params) pages.
 
+In addition to parameter setup, you can configure the simulation perfomance setup (timeout, CPU, GPU and Memory) as well.
+This parameters can be found in ```.citros/simulations/simulation_turtlebot3.json```. <br/>
+The default setup is 600 seconds timeout, 4 CPU, 4 GPU and 4096 MB of Memory.
```
[the remark this answers is not in the document]
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md]: fixed

File: docs_tutorials/turtlebot/turtlebot.md [hunk 16]
```diff
@@ -3,233 +3,150 @@ sidebar_position: 50
...
-
-### CITROS Web Usage
-#### Launching Project via CITROS Web
-The best way to use all the innovative capabilities of CITROS is through it's Web interface. Follow [this manual](https://citros.io/doc/docs/simulations/sim_overview) to easily launch a simulation on CITROS Web platform.
+All the results will be saved under .citros/runs/[simulation_name] folder.
 
-#### Working with Integrated Jupiter Notebooks and Data Analysis
-CITROS Web provides a powerful data analysis package, which is a comprehensive solution for data query, analysis and visualization. With its extensive features, you can quickly and easily extract valuable insights from your data. To use it, Jupiter Notebook support is built-in. 
-Navigate to our ```Code``` project page, open the Notebooks folder and click on the notebook file. Here you can see the usual Jupiter editor interface: you can add blocks of code or built-in Markdown engine, run and save notebook and control the Python kernel.
+To plot the local run results you can use [Foxglove](https://citros.io/doc/docs_tutorials/#visualization-with-foxglove) with the ```layout.json``` layout that exists in the ```Foxglove_layouts``` directory.
```
[the remark this answers is not in the document]
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md:119]: done

**Comment by gtep96** [comment]
@mariaBer i've updated the tutorial, but not is the scenario description clear enough now, just wanted to discuss it with you. anyway you can check it

**Comment by gtep96** [comment]
updated and pushed

File: docs_tutorials/turtlebot/turtlebot.md [hunk 17]
```diff
@@ -3,233 +3,140 @@ sidebar_position: 50
...
+This integration allows users to observe how the robot behaves in a controlled environment.<br/>
+The node includes a simple Autonomous Collision Avoidance module which ensures the TurtleBot3 avoids collisions with obstacles in its path, it utilizes sensor data, such as simulated lidar readings, to detect obstacles and adjust the robot's trajectory accordingly. The output of the simulation is a robot's trajectory, sensors outputs and other [standart TurtleBot3 ROS 2 topics](https://emanual.robotis.com/docs/en/software/robotis_framework_packages/).<br/>
+The project gives users the flexibility to modify various TurtleBot3 specifications by changing ROS 2 parameters. These parameters may include attributes like the robot's size, speed, sensor range, or collision avoidance algorithms. Users can experiment with different parameter values to observe their impact on collision avoidance behavior. The TurtleBot3 robot is capable of orienting itself effectively within a prepared TurtleBot3 world. It utilizes the simulated lidar module to gather environmental data, allowing it to make informed decisions about its navigation path.<br/>
```
Comment (mariaBer) [inline docs_tutorials/turtlebot/turtlebot.md:65]: This part until for this example should not be here

File: docs_tutorials/turtlebot/turtlebot.md [hunk 18]
```diff
@@ -3,233 +3,140 @@ sidebar_position: 50
...
+For this example, let's check how the powerful CITROS Error Analysis (a part of Data Analysis) package works. We will analyse data from IMU (Inertial Measurement Unit), Odometer and position information. To do it out, we need to set up parameters and launch CITROS simulation.<br/>
+The parameter setup is listed in ```.citros/parameter_setups/default_param_setup.json```:
```
Comment (mariaBer) [inline docs_tutorials/turtlebot/turtlebot.md:67]: In this tutorial, we will examine the powerful CITROS Error Analysis (a part of Data Analysis) package. We will analyze data from IMU (Inertial Measurement Unit), Odometer and position information. For that, we need to set up parameters and launch CITROS simulation. The parameter setup is listed in .citros/parameter_setups/default_param_setup.json:

**Review by mariaBer — CHANGES_REQUESTED** [review]

File: docs_tutorials/turtlebot/turtlebot.md (hunk 17 above)
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md]: Done

File: docs_tutorials/turtlebot/turtlebot.md (hunk 18 above)
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md]: Done

**Comment by gtep96** [comment]
Done and pushed

File: docs_tutorials/turtlebot/turtlebot.md [hunk 19]
```diff
@@ -7,6 +7,7 @@ sidebar_label: 'TurtleBot'
 
 ## Overview
 This project is designed to wrap the official [TurtleBot3 simulation example](https://emanual.robotis.com/docs/en/platform/turtlebot3/simulation/#gazebo-simulation) and integrate it with a simple Autonomous Collision Avoidance node. The node provides ROS 2 parameters that allow users to adjust various TurtleBot3 specifications, enabling the analysis of how these changes affect collision avoidance behavior. <br />
```
Comment (mariaBer) [inline docs_tutorials/turtlebot/turtlebot.md:9]: Is this part needed if you added the section below?
The node provides ROS 2 parameters that allow users to adjust various TurtleBot3 specifications, enabling the analysis of how these changes affect collision avoidance behavior. <br />

**Review by mariaBer — CHANGES_REQUESTED** [review]

File: docs_tutorials/turtlebot/turtlebot.md (hunk 19 above)
Comment (gtep96) [reply docs_tutorials/turtlebot/turtlebot.md]: I think it's possible to delete the sentence "The node provides ROS 2 parameters that allow users to adjust various TurtleBot3 specifications, enabling the analysis of how these changes affect collision avoidance behavior." Done

**Review by mariaBer — APPROVED** [review]

**Pull request merged**